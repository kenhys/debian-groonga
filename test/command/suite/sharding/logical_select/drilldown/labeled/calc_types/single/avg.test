#@on-error omit
plugin_register sharding
#@on-error default

table_create Tags TABLE_PAT_KEY ShortText

table_create Memos_20150709 TABLE_HASH_KEY ShortText
column_create Memos_20150709 created_at COLUMN_SCALAR Time
column_create Memos_20150709 tag COLUMN_SCALAR Tags
column_create Memos_20150709 priority COLUMN_SCALAR Int64

table_create Memos_20150710 TABLE_HASH_KEY ShortText
column_create Memos_20150710 created_at COLUMN_SCALAR Time
column_create Memos_20150710 tag COLUMN_SCALAR Tags
column_create Memos_20150710 priority COLUMN_SCALAR Int64

load --table Memos_20150709
[
{"_key": "Groonga1", "tag": "Groonga", "priority": 10,
 "created_at": "2015/07/09 00:00:00"},
{"_key": "Groonga2", "tag": "Groonga", "priority": 20,
 "created_at": "2015/07/09 01:00:00"},
{"_key": "Groonga3", "tag": "Groonga", "priority": 60,
 "created_at": "2015/07/09 02:00:00"},
{"_key": "Mroonga1", "tag": "Mroonga", "priority": 61,
 "created_at": "2015/07/09 03:00:00"}
]

load --table Memos_20150710
[
{"_key": "Mroonga2", "tag": "Mroonga", "priority": 24,
 "created_at": "2015/07/10 00:00:00"},
{"_key": "Mroonga3", "tag": "Mroonga", "priority": 8,
 "created_at": "2015/07/10 01:00:00"},
{"_key": "Rroonga1", "tag": "Rroonga", "priority": 3,
 "created_at": "2015/07/10 02:00:00"},
{"_key": "Rroonga2", "tag": "Rroonga", "priority": -9,
 "created_at": "2015/07/10 03:00:00"},
{"_key": "Rroonga3", "tag": "Rroonga", "priority": 0,
 "created_at": "2015/07/10 04:00:00"}
]

logical_select Memos \
  --shard_key created_at \
  --limit 0 \
  --drilldown[tag].keys tag \
  --drilldown[tag].calc_types AVG \
  --drilldown[tag].calc_target priority \
  --drilldown[tag].output_columns _key,_avg

