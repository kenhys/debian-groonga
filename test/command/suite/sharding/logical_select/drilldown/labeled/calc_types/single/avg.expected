plugin_register sharding
[[0,0.0,0.0],true]
table_create Tags TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
table_create Memos_20150709 TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos_20150709 created_at COLUMN_SCALAR Time
[[0,0.0,0.0],true]
column_create Memos_20150709 tag COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
column_create Memos_20150709 priority COLUMN_SCALAR Int64
[[0,0.0,0.0],true]
table_create Memos_20150710 TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos_20150710 created_at COLUMN_SCALAR Time
[[0,0.0,0.0],true]
column_create Memos_20150710 tag COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
column_create Memos_20150710 priority COLUMN_SCALAR Int64
[[0,0.0,0.0],true]
load --table Memos_20150709
[
{"_key": "Groonga1", "tag": "Groonga", "priority": 10,
 "created_at": "2015/07/09 00:00:00"},
{"_key": "Groonga2", "tag": "Groonga", "priority": 20,
 "created_at": "2015/07/09 01:00:00"},
{"_key": "Groonga3", "tag": "Groonga", "priority": 60,
 "created_at": "2015/07/09 02:00:00"},
{"_key": "Mroonga1", "tag": "Mroonga", "priority": 61,
 "created_at": "2015/07/09 03:00:00"}
]
[[0,0.0,0.0],4]
load --table Memos_20150710
[
{"_key": "Mroonga2", "tag": "Mroonga", "priority": 24,
 "created_at": "2015/07/10 00:00:00"},
{"_key": "Mroonga3", "tag": "Mroonga", "priority": 8,
 "created_at": "2015/07/10 01:00:00"},
{"_key": "Rroonga1", "tag": "Rroonga", "priority": 3,
 "created_at": "2015/07/10 02:00:00"},
{"_key": "Rroonga2", "tag": "Rroonga", "priority": -9,
 "created_at": "2015/07/10 03:00:00"},
{"_key": "Rroonga3", "tag": "Rroonga", "priority": 0,
 "created_at": "2015/07/10 04:00:00"}
]
[[0,0.0,0.0],5]
logical_select Memos   --shard_key created_at   --limit 0   --drilldown[tag].keys tag   --drilldown[tag].calc_types AVG   --drilldown[tag].calc_target priority   --drilldown[tag].output_columns _key,_avg
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        9
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "created_at",
          "Time"
        ],
        [
          "priority",
          "Int64"
        ],
        [
          "tag",
          "Tags"
        ]
      ]
    ],
    {
      "tag": [
        [
          3
        ],
        [
          [
            "_key",
            "ShortText"
          ],
          [
            "_avg",
            "Float"
          ]
        ],
        [
          "Groonga",
          30.0
        ],
        [
          "Mroonga",
          31.0
        ],
        [
          "Rroonga",
          -2.0
        ]
      ]
    }
  ]
]
